### PR TITLE
MenuBar changes for #4463

### PR DESCRIPTION
**Pull request by smorin4**
# PR checklist

Changed menu bar to include animated settings button on top right (with full navigation to settings page) when settings is chosen in wizard.

Applies to:
#4463 

**Applies to the following platforms:**

- [x] WinUI
- [ ] WPF
- [ ] UWP

### CODE REVIEW

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml [hunk 1]
```diff
@@ -21,27 +22,28 @@
...
-                <!--
-                TODO: You can edit the text for the menu in String/en-US/Resources.resw
-                Adding <MenuBar.Items> xaml code is a workaround to the following issue
-                https://github.com/microsoft/microsoft-ui-xaml/issues/4385
-                -->
-                <MenuBar.Items>
+            <Grid Grid.Row="0">
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="*" />
+                    <ColumnDefinition Width="Auto" />
```
Comment (mrlacey) [inline code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml:31]: What's this column (definition) for? I can't see why it exists

File: code/TemplateStudioForWinUICs/TemplateStudioForWinUICs.csproj [hunk 2]
```diff
@@ -212,6 +212,9 @@
       <DependentUpon>source.extension.vsixmanifest</DependentUpon>
     </Compile>
     <Compile Include="TemplateStudioForWinUICsPackage.cs" />
+    <Content Include="Templates\_comp\MT\Project.MenuBar.AddNavViewItem.Settings\Views\ShellPage_postaction.xaml.cs">
```
Comment (mrlacey) [inline code/TemplateStudioForWinUICs/TemplateStudioForWinUICs.csproj:215]: How did this not being included previously not break something?
If, as it appears, this was not included but should have been it should have caused something to break. It's not enough just to fix this, we need to ensure that a similar situation can't happen again.

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml [hunk 3]
```diff
@@ -1,7 +1,16 @@
-﻿                    <MenuBarItem x:Uid="ShellMenuBarItem_File">
-                        <MenuBarItem.Items>
+﻿                </MenuBar>
 <!--{[{-->
-                            <MenuFlyoutItem x:Uid="ShellMenuItem_File_Param_ItemName" Command="{x:Bind ViewModel.MenuFileParam_ItemNameCommand}" />
+                <AppBarButton Grid.Column="1" x:Name="SettingsButton" AnimatedIcon.State="Normal"
```
Comment (mrlacey) [inline code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml:3]: The naming of this template is all wrong. I know it existed before this PR but now is a good time to fix it.
The name refers to a "NavViewItem" but this template is only applicable to MenuBar projects and so the reference to a NavigationView(Item) is misleading and confusing. (I wasted several minutes being confused about where/when/how this applies.)

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml [hunk 4]
```diff
@@ -21,27 +22,28 @@
         </SplitView.Pane>
         <Grid>
             <Grid.RowDefinitions>
-                <RowDefinition Height="Auto" />
+                <RowDefinition Height="48" />
```
Comment (mrlacey) [inline code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml:25]: Where did this "magic number" come from?
General rule: Don't use "magic numbers" they make maintenance harder and more time consuming.

**Review by mrlacey — CHANGES_REQUESTED** [review]
As per inline comments

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml (hunk 1 above)
Comment (mikebattista) [reply code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml:31]: It's for the Settings button when the SettingsPage is added.

File: code/TemplateStudioForWinUICs/TemplateStudioForWinUICs.csproj (hunk 2 above)
Comment (mikebattista) [reply code/TemplateStudioForWinUICs/TemplateStudioForWinUICs.csproj:215]: This is a net new file.

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml (hunk 3 above)
Comment (mikebattista) [reply code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml:3]: I would like to clean up template names like this as well as rename SplitView to NavigationView throughout the templates. I'd rather we get these and other changes in first, then we can rename everything. I've filed #4498 to track the renames.

Comment (mrlacey) [inline code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml:3]: Accessibility Insights is complaining about this button not having an accessible name. 
Is this an issue with the underlying control or the way this has been implemented?

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml (hunk 1 above)
Comment (mikebattista) [reply code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar/Views/ShellPage.xaml:31]: Thanks. Yes we considered making the composition template more complex but preferred this simpler approach as well.

File: code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml (hunk 3 above)
Comment (mikebattista) [reply code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml:3]: What defines an accessible name?

Comment (mikebattista) [reply code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml:3]: I tried setting AutomationProperties.Name as well as x:Uid in addition to x:Name which was already there, but the tool still raises the error. The provided docs suggest setting AutomationProperties.Name on the AppBarButton should resolve the issue.

Comment (mikebattista) [reply code/TemplateStudioForWinUICs/Templates/_comp/MT/Project.MenuBar.AddNavViewItem.Settings/Views/ShellPage_postaction.xaml]: Fixed.

**Review by mikebattista — APPROVED** [review]

**Comment by mikebattista** [comment]
I'm going to merge this since the remaining open issue will be addressed as part of the title bar review where we are getting specs from the WinUI team on where we need custom styles and what those styles should be.

**Comment by mikebattista** [comment]
Opened #4510 to track remaining issues with theming after removing custom styles.

**Pull request merged**